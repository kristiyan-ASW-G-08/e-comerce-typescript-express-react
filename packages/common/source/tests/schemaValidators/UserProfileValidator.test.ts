import UserProfileValidator from '@schemaValidators/UserProfileValidator';

describe('UserProfileValidator', (): void => {
  const validUsers = [
    {
      username: 'validUsername',
      handle: 'testUserHandle',
      website: 'https://testwebsite.test',
    },
    {
      username: 'Barbatos',
      handle: 'LupusREx',
      website: 'http://somewebsite.com',
    },
  ];
  const invalidUsers = [
    {
      username: '',
      handle: '',
      website: 'testwebsite.test',
    },
  ];
  it.each(validUsers)('should validate successfully', async user => {
    expect.assertions(1);
    await expect(
      UserProfileValidator.validate(user, { abortEarly: false }),
    ).resolves.toBe(user);
  });
  it.each(invalidUsers)('should validate unsuccessfully', async user => {
    expect.assertions(1);
    await expect(
      UserProfileValidator.validate(user, { abortEarly: false }),
    ).rejects.toMatchSnapshot();
  });
});
