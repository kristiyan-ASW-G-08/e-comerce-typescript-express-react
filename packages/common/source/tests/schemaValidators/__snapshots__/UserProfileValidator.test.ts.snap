// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileValidator should validate unsuccessfully 1`] = `[ValidationError: 4 errors occurred]`;
